hermes-consumers/src/main/java/pl/allegro/tech/hermes/consumers/consumer/receiver/kafka/KafkaMessageReceiverFactory.java

```diff
@@ -98,7 +101,11 @@ public MessageReceiver createMessageReceiver(Topic topic,
         props.put(SESSION_TIMEOUT_MS_CONFIG, configs.getIntProperty(Configs.KAFKA_CONSUMER_SESSION_TIMEOUT_MS_CONFIG));
         props.put(HEARTBEAT_INTERVAL_MS_CONFIG, configs.getIntProperty(Configs.KAFKA_CONSUMER_HEARTBEAT_INTERVAL_MS_CONFIG));
         props.put(METADATA_MAX_AGE_CONFIG, configs.getIntProperty(Configs.KAFKA_CONSUMER_METADATA_MAX_AGE_CONFIG));
-        props.put(MAX_PARTITION_FETCH_BYTES_CONFIG, configs.getIntProperty(Configs.KAFKA_CONSUMER_MAX_PARTITION_FETCH_BYTES_CONFIG));
+
+        SubscriptionFetchingConfiguration fetchingConfiguration = subscription.getFetchingConfiguration();
+        int globalMaxBytes = configs.getIntProperty(Configs.KAFKA_CONSUMER_MAX_PARTITION_FETCH_BYTES_CONFIG);
+        props.put(MAX_PARTITION_FETCH_BYTES_CONFIG, min(fetchingConfiguration.getMaxPartitionFetchBytes(), globalMaxBytes));
```

Comment:
why min? isn't the whole point of it to be able to increase the global size? maybe we should have default (1Mb) and max_max (like 10Mb)